liblangutil/SourceReferenceExtractor.h

```diff
@@ -31,28 +31,27 @@ namespace langutil
 
 struct LineColumn
 {
-	int line;
-	int column;
+	int line {-1};
+	int column {-1};
 
+	LineColumn() = default;
 	LineColumn(std::tuple<int, int> const& _t): line{std::get<0>(_t)}, column{std::get<1>(_t)} {}
-	LineColumn(int _line, int _column): line{_line}, column{_column} {}
-	LineColumn(): line{-1}, column{-1} {}
 };
 
 struct SourceReference
 {
-	std::string message;    ///< A message that relates to this source reference (such as a warning or an error message).
-	std::string sourceName; ///< Underlying source name (for example the filename).
-	LineColumn position;    ///< Actual (error) position this source reference is surrounding.
-	bool multiline;         ///< Indicates whether the actual SourceReference is truncated to one line.
-	std::string text;       ///< Extracted source code text (potentially truncated if multiline or too long).
-	int startColumn;        ///< Highlighting range-start of text field.
-	int endColumn;          ///< Highlighting range-end of text field.
+	std::string message {};    ///< A message that relates to this source reference (such as a warning or an error message).
```

Comment:
@ekpyron Actually it does make sense the way I wrote it, here, because it eliminates the need of constructors as long as you don't need custom logic in the constructor bodies.

Reply:
Are you sure? I'm pretty certain that it'd be *exactly* the same, if you would just use ``std::string message;`` - in both cases it will default construct even without defining a constructor. You only need to member-initialize for PODs or types that are not default-constructible.

Reply:
Or if you don't want the member to be default-constructed of course :-).